### PR TITLE
upgrade bs-platform to 4.0.6

### PR DESCRIPTION
**Pull request by joprice**

### CODE REVIEW

File: bsconfig.json [hunk 1]
```diff
@@ -2,6 +2,10 @@
   "name": "bs-fetch",
   "bsc-flags": ["-bs-no-version-header", "-bs-super-errors"],
   "refmt": 3,
+  "package-specs": [{
+    "module": "commonjs",
+    "in-source": false
+  }],
```
Comment (glennsl) [inline bsconfig.json:8]: Can you explain why this is needed? Isn't this the default?

**Review by glennsl — COMMENTED** [review]
Thanks! I removed yarn.lock though, since I usually use npm and a lock file doesn't matter much for this kind of project anyway.

File: bsconfig.json (hunk 1 above)
Comment (joprice) [reply bsconfig.json:8]: I can run it again locally without it. I remembering needing to add it because it generated the files in-source.

Comment (joprice) [reply bsconfig.json:8]: Just tried it and it worked without. Not sure what gave me the wrong impression there.

**Comment by joprice** [comment]
Isn't it best practice to use a lock file for yarn or npm? If not and you prefer not to use them, should it be in the gitignore so other contributors don't redo this mistake?

File: bsconfig.json (hunk 1 above)
Comment (glennsl) [reply bsconfig.json:8]: Nah, don't worry about it. It's better to be explicit anyway, I was just wondering why. I guess the default's changed, and it's not the first undocumented breaking change in a patch release coming from bucklescript.

Comment (joprice) [reply bsconfig.json:8]: I moved too fast. Already changed it. And Im already on the subway headed to my brothers wedding. I won’t be back until Monday. I can fix then or feel free to abandon and redo. I’m not worried about contributor stats just want to silence a warning :-)

**Comment by glennsl** [comment]
The only dependency is bs-platform, which is a build tool, not a "normal" dependency, a devDependency that is overridden by consuming projects, and is expected by its author to always be the latest anyway. And even globally installed and linked into projects.

It doesn't matter much if it's included in PRs either, since it's easy enough to remove. And whether it's gitignored or npms lock file is committed doesn't matter much to me. But yarn.lock would go stale since I don't use it.

**Comment by glennsl** [comment]
If you add the specs back, or confirm it works without them, I'll merge.

**Comment by joprice** [comment]
That makes sense. I'll restore the setting to be explicit about it.

**Comment by mlms13** [comment]
Didn't see this PR here, but when it's merged, it will fix #20.

File: bsconfig.json (hunk 1 above)
Comment (joprice) [reply bsconfig.json:8]: @glennsl I restored the config

**Comment by glennsl** [comment]
Thank you!

**Pull request merged**